.travis.yml

```diff
@@ -7,30 +7,24 @@ dist: trusty
 
 env:
   matrix:
-    - xGSL=0 xMPI=0 xPYTHON=0
-    - xGSL=0 xMPI=0 xPYTHON=1
-    - xGSL=0 xMPI=1 xPYTHON=0
-    - xGSL=0 xMPI=1 xPYTHON=1
-    - xGSL=1 xMPI=0 xPYTHON=0
-    - xGSL=1 xMPI=0 xPYTHON=1
-    - xGSL=1 xMPI=1 xPYTHON=0
-    - xGSL=1 xMPI=1 xPYTHON=1
+    - xGSL=0 xMPI=0 xPYTHON=1 CACHE_NAME=JOB
```

Comment:
You can have a different cache for each job. Giving a `CACHE_NAME` ensures, that we only have one cache. See this documentation about [TravisCI caching](https://docs.travis-ci.com/user/caching/#Caches-and-build-matrices) (point 4. would make different caches in our case).